code/espurna/thinkspeak.ino

```diff
@@ -256,55 +272,95 @@ void _tspkPost() {
 
 #else // THINGSPEAK_USE_ASYNC
 
-void _tspkPost() {
+#if THINGSPEAK_USE_SSL && (SECURE_CLIENT == SECURE_CLIENT_BEARSSL)
+
+SecureClientConfig _tspk_sc_config {
+    "THINGSPEAK",
+    []() -> int {
+        return getSetting("tspkScCheck", THINGSPEAK_SECURE_CLIENT_CHECK);
+    },
+    []() -> PGM_P {
+        return _tspk_client_trusted_root_ca;
+    },
+    []() -> String {
+        return getSetting("tspkFP", THINGSPEAK_FINGERPRINT);
+    },
+    []() -> uint16_t {
+        return getSetting("tspkScMFLN", THINGSPEAK_SECURE_CLIENT_MFLN);
+    },
+    true
+};
 
-    #if THINGSPEAK_USE_SSL
-        WiFiClientSecure _tspk_client;
-    #else
-        WiFiClient _tspk_client;
-    #endif
+#endif // THINGSPEAK_USE_SSL && SECURE_CLIENT_BEARSSL
 
-    if (_tspk_client.connect(_tspk_host.c_str(), _tspk_port)) {
+void _tspkPost(WiFiClient* client, const URL& url) {
 
-        DEBUG_MSG_P(PSTR("[THINGSPEAK] Connected to %s:%u\n"), _tspk_host.c_str(), _tspk_port);
+    if (!client->connect(url.host.c_str(), url.port)) {
+        DEBUG_MSG_P(PSTR("[THINGSPEAK] Connection failed\n"));
+        return;
+    }
 
-        if (!_tspk_client.verify(THINGSPEAK_FINGERPRINT, _tspk_host.c_str())) {
-            DEBUG_MSG_P(PSTR("[THINGSPEAK] Warning: certificate doesn't match\n"));
-        }
+    DEBUG_MSG_P(PSTR("[THINGSPEAK] Connected to %s:%u\n"), url.host.c_str(), url.port);
+    DEBUG_MSG_P(PSTR("[THINGSPEAK] POST %s?%s\n"), url.path.c_str(), _tspk_data.c_str());
 
-        DEBUG_MSG_P(PSTR("[THINGSPEAK] POST %s?%s\n"), _tspk_client.path.c_str(), _tspk_data.c_str());
-        char headers[strlen_P(THINGSPEAK_REQUEST_TEMPLATE) + _tspk_client.path.length() + _tspk_client.host.lengh() + 1];
-        snprintf_P(headers, sizeof(headers),
-            THINGSPEAK_REQUEST_TEMPLATE,
-            _tspk_client.path.c_str(),
-            _tspk_client.host.c_str(),
-            _tspk_data.length()
-        );
+    char headers[strlen_P(THINGSPEAK_REQUEST_TEMPLATE) + url.path.length() + url.host.length() + 1];
+    snprintf_P(headers, sizeof(headers),
+        THINGSPEAK_REQUEST_TEMPLATE,
+        url.path.c_str(),
+        url.host.c_str(),
+        _tspk_data.length()
+    );
 
-        _tspk_client.print(headers);
-        _tspk_client.print(_tspk_data);
+    client->print(headers);
+    client->print(_tspk_data);
 
-        nice_delay(100);
+    nice_delay(100);
 
-        String response = _tspk_client.readString();
-        int pos = response.indexOf("\r\n\r\n");
-        unsigned int code = (pos > 0) ? response.substring(pos + 4).toInt() : 0;
-        DEBUG_MSG_P(PSTR("[THINGSPEAK] Response value: %u\n"), code);
-        _tspk_client.stop();
+    const auto response = client->readString();
+    int pos = response.indexOf("\r\n\r\n");
```

Comment:
This whole part should be HTTPClient, since we have it readily available from `#include <ESP8266HTTPClient.h>`
At least we will know about the status code.